salt/states/dockerng.py

```diff
@@ -202,6 +202,15 @@ def _image_get(path, default=None):
                     env_ptr = env_diff.setdefault(item, {})
                     env_ptr.setdefault('old', {})[key] = actual_val
                     env_ptr.setdefault('new', {})[key] = data[key]
+            # Comparing list vice-versa
+            for key in actual_env:
+                # Need to skip PATH and no_proxy variables.
+                if key not in ['PATH', 'no_proxy']:
```

Comment:
Please add in the comment that those environment variables are controlled by docker then.